awx/main/scheduler/task_manager.py

```diff
@@ -626,15 +629,19 @@ def process_pending_tasks(self, pending_tasks):
 
             # All task.capacity_type == 'control' jobs should run on control plane, no need to loop over instance groups
             if task.capacity_type == 'control':
+                if not self.instance_groups[self.controlplane_ig.name].has_remaining_capacity(control_impact=True):
+                    continue
                 task.execution_node = control_instance.hostname
-                execution_instance = self.instances[control_instance.hostname].obj
+                execution_instance = self.instance_groups.task_manager_instances[control_instance.hostname].obj
                 task.log_lifecycle("controller_node_chosen")
                 task.log_lifecycle("execution_node_chosen")
                 self.start_task(task, self.controlplane_ig, task.get_jobs_fail_chain(), execution_instance)
                 found_acceptable_queue = True
                 continue
 
             for instance_group in self.instance_groups.get_instance_groups_from_task_cache(task):
+                if not self.instance_groups[instance_group.name].has_remaining_capacity(task):
+                    continue
```

Comment:
I do think we should have a log, like:

```python
                    logger.debug(
                        f"Instance group {instance_group.name} not available to run job {task.log_format} w/ capacity requirement {task.task_impact}"
                    )
```

Reply:
I did add the log in the `has_remaining_capacity` though probably should update to use `task.log_format` i'll comment there to point it out

Reply:
pointed out later where I do log in the case that we max_forks or max_concurrent_jobs is blocking (only at debug level)